types/context.go

```diff
@@ -60,6 +62,8 @@ func (c Context) GasMeter() storetypes.GasMeter                 { return c.gasMe
 func (c Context) BlockGasMeter() storetypes.GasMeter            { return c.blockGasMeter }
 func (c Context) IsCheckTx() bool                               { return c.checkTx }
 func (c Context) IsReCheckTx() bool                             { return c.recheckTx }
+func (c Context) IsPrepareProposal() bool                       { return c.prepareProposal }
```

Comment:
is this meant to be used within modules?

Reply:
I can undo the Context stuff in this PR if you want further consideration or discussion out of band